libsql/sql.go

```diff
@@ -37,6 +37,26 @@ func extractJwt(u *url.URL) (string, error) {
 	}
 }
 
+func extractTls(query *url.Values, scheme string) (bool, error) {
+	tls := query.Get("tls")
+	query.Del("tls")
+	if tls == "" {
+		if scheme == "http" || scheme == "ws" {
+			tls = "0"
+		} else {
+			tls = "1"
+		}
+	}
+
+	if tls == "0" {
+		return false, nil
+	} else if tls == "1" {
+		return true, nil
+	} else {
+		return true, fmt.Errorf("unknown value of tls query parameter. Valid values are 0 and 1")
```

Comment:
I think `false` is the zero-value for bool, no?

Reply:
Because using TLS is the "safe default"; on the remote chance that the caller does not check for the error, it's safer if it attempts to use TLS.

Reply:
It seems non-idiomatic but fair